File: 백준/Silver/13305. 주유소/주유소.py
import sys
n=int(sys.stdin.readline())
distances=list(map(int,sys.stdin.readline().split()))
prices=list(map(int,sys.stdin.readline().split()))
minprice,money=prices[0],0
for i in range(1,n):
    money+=minprice*distances[i-1]
    if(minprice>=prices[i]):
        minprice=prices[i]
print(money)

File: 백준/Silver/13305. 주유소/주유소.c
#include <stdio.h>
#include <stdlib.h>
int main(void) {
    int n, elem, minprice, money = 0;
    scanf("%d", &n);
    int *distances = (int *)malloc(sizeof(int) * (n - 1));
    int *prices = (int *)malloc(sizeof(int) * n);
    for (int i = 0; i < n - 1; i++) {
        scanf("%d", &elem);
        distances[i] = elem;
    }
    for (int i = 0; i < n; i++) {
        scanf("%d", &elem);
        prices[i] = elem;
    }
    minprice = prices[0];
    for (int i = 1; i < n; i++) {
        money += minprice * distances[i - 1];
        if (minprice >= prices[i])
            minprice = prices[i];
    }
    printf("%d", money);
    return 0;
}